enterprise/utils.py

```diff
@@ -123,3 +124,27 @@ def patch_mako_lookup():
     # Both add an item to the setting AND insert the lookup for immediate use
     settings.MAKO_TEMPLATES['main'].insert(0, template_location)
     add_lookup('main', template_location)
+
+
+def get_catalog_admin_url_template():
+    """
+    Get template of catalog admin url.
+
+    URL template will contain a placeholder '{catalog_id}' for catalog id.
+
+    Example:
+        >>> get_catalog_admin_url_template()
+        "http://localhost:18381/admin/catalogs/catalog/{catalog_id}/change/"
+
+    Returns:
+        A string containing template for catalog url.
+    """
+    api_base_url = getattr(settings, "COURSE_CATALOG_API_URL", "")
+
+    # Extract FQDN (Fully Qualified Domain Name) from API URL.
+    match = re.match(r"^(?P<fqdn>(?:https?://)?[^/]+)", api_base_url)
+
+    if match:
+        return match.group("fqdn").rstrip("/") + "/admin/catalogs/catalog/{catalog_id}/change/"
```

Comment:
nit: add a comment here, what is returning here etc etc.. to better understanding.